vcpkg/maintainers/functions/vcpkg_make_configure.md

```diff
@@ -0,0 +1,228 @@
+  - `--libdir` : Subdirectory under the prefix for libraries (shared or static)
+  - `--mandir` : Subdirectory under the prefix for manual pages (man files)
+  - `--docdir` : Subdirectory under the prefix for documentation
+  - `--datarootdir` : Subdirectory under the prefix for architecture-independent data
+
+* Debug builds only:
+  - `--includedir` : Subdirectory under the prefix for header files
+
+Example Paths:
+* Release Configuration:
+  - `--prefix=${current_installed_dir}`
+  - `--bindir=${prefix}/tools/${PORT}/bin`
+  - `--sbindir=${prefix}/tools/${PORT}/sbin`
+  - `--libdir=${prefix}/lib`
+  - `--mandir=${prefix}/share/${PORT}`
+  - `--docdir=${prefix}/share/${PORT}`
+  - `--datarootdir=${prefix}/share/${PORT}`
+
+* Debug Configuration:
+  - `--prefix=${current_installed_dir}${path_suffix_DEBUG}`
+  - `--bindir=${prefix}/../tools/${PORT}${path_suffix_DEBUG}/bin`
+  - `--sbindir=${prefix}/../tools/${PORT}${path_suffix_DEBUG}/sbin`
+  - `--libdir=${prefix}/lib`
+  - `--includedir=${prefix}/../include`
+  - `--mandir=${prefix}/share/${PORT}`
+  - `--docdir=${prefix}/share/${PORT}`
+  - `--datarootdir=${prefix}/share/${PORT}`
+
+Platform-Specific Tweaks:
+* On Windows
+  - Sets `gl_cv_double_slash_root=yes` to signal to the configure script that it is running on Windows, where paths can begin with double slashes (//).
+  - Sets `ac_cv_func_memmove=yes` and `ac_cv_func_memset=yes` to indicate that the `memmove` and `memset` functions are available.
+  - sets `gl_cv_host_cpu_c_abi=no` on ARM64 builds to indicate that the compiler does not use the standard C ABI for floating-point numbers.
+
+* On Unix-like systems:
+  - Adjusts paths to match the typical Unix filesystem hierarchy (e.g., using /usr/local).
+
+>[!NOTE]
+> These default options streamline the configuration process for most projects. If you need a highly customized build, you can set `DISABLE_DEFAULT_OPTIONS` and manually provide all the necessary configuration options.
+
+### OPTIONS
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+### OPTIONS_RELEASE
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+These are in addition to `OPTIONS`.
+
+### OPTIONS_DEBUG
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+These are in addition to `OPTIONS`.
+
+### PRE_CONFIGURE_CMAKE_COMMANDS
+
+Specifies a list of CMake commands to execute before the configure process for each build type. That is, if `VCPKG_BUILD_TYPE` is empty (the default), these commands will run once for debug and once for release. If `VCPKG_BUILD_TYPE` is `release`, these commands will only be once for release. This can be used to set up the environment or prepare the build in a way that the configure script requires.
```

Comment:
```suggestion
Specifies a list of CMake commands to execute before the configure process for each build type. That is, if `VCPKG_BUILD_TYPE` is empty (the default), these commands are run once for debug and once for release. If `VCPKG_BUILD_TYPE` is `release`, these commands are only run once for release. This can be used to set up the environment or prepare the build in a way that the configure script requires.
```